processor/src/errors.rs

```diff
@@ -28,3 +30,11 @@ pub enum ExecutionError {
     SyscallTargetNotInKernel(Digest),
     UnexecutableCodeBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(fmt, "{self:?}")
```

Comment:
It must be done, but we normally add a more human-friendly description to our error `Display::fmt` implementations. Something like

`DuplicateAdviceKey(key) => write!(fmt, "an attempt to insert a duplicated advice key {key:?} was made"),`

Reply:
https://github.com/0xPolygonMiden/miden-vm/pull/688#discussion_r1100126794

I'm open to suggestions of text. I could also do a best effort since some of those I can guess what  they are doing, but others like `UninitializedMemoryAddress` I have no idea why it happens and what the error means.

Reply:
I did a best effort, should be fine now.